lib/Arch/SPARC32_C.cpp

```diff
@@ -0,0 +1,347 @@
+    case llvm::Type::PointerTyID: {
+      ret_values.emplace_back();
+      auto &value_declaration = ret_values.back();
+      value_declaration.reg = arch->RegisterByName("o0");
+      value_declaration.type = ret_type;
+      return llvm::Error::success();
+    }
+
+    case llvm::Type::HalfTyID:
+    case llvm::Type::FloatTyID: {
+      ret_values.emplace_back();
+      auto &value_declaration = ret_values.back();
+      value_declaration.reg = arch->RegisterByName("f0");
+      value_declaration.type = ret_type;
+      return llvm::Error::success();
+    }
+
+    case llvm::Type::DoubleTyID: {
+      ret_values.emplace_back();
+      auto &value_declaration = ret_values.back();
+      value_declaration.reg = arch->RegisterByName("d0");
+      value_declaration.type = ret_type;
+      return llvm::Error::success();
+    }
+
+    // Try to split the composite type over registers, and fall back on RVO
+    // if it's not possible.
+    case llvm::GetFixedVectorTypeId():
+    case llvm::Type::ArrayTyID:
+    case llvm::Type::StructTyID: {
+      AllocationState alloc_ret(return_register_constraints, arch, this);
+      alloc_ret.config.type_splitter = IntegerTypeSplitter;
+      auto mapping = alloc_ret.TryRegisterAllocate(*ret_type);
+
+      // There is a valid split over registers, so add the mapping
+      if (mapping) {
+        return alloc_ret.CoalescePacking(mapping.getValue(), ret_values);
+
+      // Composite type splitting; Unlike with x86, LLVM doesn't naturally
+      // perform RVO on large structures returned by value from bitcode.
+      } else {
+        break;
+      }
+    }
+
+    default:
+      break;
+  }
+
+  return llvm::createStringError(
+      std::errc::invalid_argument,
+      "Could not allocate unsupported type '%s' to return register in function '%s'",
+      remill::LLVMThingToString(ret_type).c_str(),
+      function.getName().str().c_str());
+}
+
+// For X86_64_SysV, the general argument passing behavior is, try to pass the
+// arguments in registers RDI, RSI, RDX, RCX, R8, R9 from integral types and
+// XMM0 - XMM7 for float types. If the argument is a struct but can be
+// completely split over the above registers, then greedily split it over the
+// registers. Otherwise, the struct is passed entirely on the stack. If we run
+// our of registers then pass the rest of the arguments on the stack.
+llvm::Error SPARC32_C::BindParameters(
+    llvm::Function &function, bool injected_sret,
+    std::vector<ParameterDecl> &parameter_declarations) {
+  CHECK(!injected_sret)
+      << "Injected struct returns are not supported on SPARC targets";
+
+  const auto param_names = TryRecoverParamNames(function);
+  llvm::DataLayout dl(function.getParent());
+
+  // Used to keep track of which registers have been allocated
+  AllocationState alloc_param(parameter_register_constraints, arch, this);
+  alloc_param.config.type_splitter = IntegerTypeSplitter;
+
+  uint64_t stack_offset = 92;
```

Comment:
Transform to constant with meaningful name -- i am curious why this is 92

Reply:
it's used once, and is assigned to a name that is semi-meaningful. The source of the number escapes me; I think I did some experiments in compiler explorer and this number popped out. It's likely something to do with the size of a register window.